Tests/Shared.Specs/StringAssertionSpecs.cs

```diff
@@ -2043,6 +2043,138 @@ public void When_the_actual_string_equivalent_to_the_expected_but_with_trailing_
 
         #endregion
 
+        #region Not Be Equivalent To
+
+        [Fact]
+        public void When_strings_are_the_same_while_ignoring_case_it_should_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            string actual = "ABC";
+            string unexpected = "abc";
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action action = () => actual.Should().NotBeEquivalentTo(unexpected);
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            action.Should().Throw<XunitException>()
+#if NETCOREAPP1_1
+                .WithMessage("Expected string not to be equivalent to \"abc\", but they are.");
+#else
+                .WithMessage("Expected actual not to be equivalent to \"abc\", but they are.");
+#endif
+        }
+
+        [Fact]
+        public void When_strings_differ_other_than_by_case_it_should_not_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => "ADC".Should().NotBeEquivalentTo("abc", "we will test {0} + {1}", 1, 2);
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_non_null_string_is_expected_to_be_equivalent_to_null_it_should_not_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => "ABCDEF".Should().NotBeEquivalentTo(null);
```

Comment:
Is this a valid case? Why would you want to compare to `null`?

Reply:
In practice you're right. I don't see why anyone would want to use this method instead of `NotBeNull()` in this scenario, but the API allows it, so might as well test for it.

If this case is not valid, then the one in the opposite direction is not valid too. https://github.com/fluentassertions/fluentassertions/blob/6dd60e7eba9d17c296e6901a9731d9fdd427bbce/Tests/Shared.Specs/StringAssertionSpecs.cs#L1954

Reply:
I would like to keep this, at least to test that no `NullReferenceException` is thrown.